components/molecules/sidebar/components/sidebar.js

```diff
@@ -3,43 +3,87 @@
  */
 
 // React
-import { array, string } from 'prop-types'
+import React, { useState } from 'react'
+import { array, string, object } from 'prop-types'
 
 // UI
 import { Image, Icon, Link, List } from '../../../'
 import { Divider } from '../../../atoms/divider/divider'
 import { ListItem } from '../../../atoms/list/components/listItem'
+import { Avatar } from 'components'
+import { Context, Size } from 'decorators'
 
 // Style
 import styled, { css } from 'styled-components'
 
-export const Sidebar = ({ brand, data }) => {
+export const Sidebar = ({ brand, data, srcAvatar, styles }) => {
+  const [isCollapsed, setIsCollapsed] = useState(false)
+
+  const handleCollapse = () => {
+    console.log('is collapsed')
+    setIsCollapsed(!isCollapsed)
+  }
+
   const link = (icon, name, to) => {
     const iconArray = Array.isArray(icon)
 
     return (
       <Link to={to} passHref>
-        <StyledLink>
+        <StyledLink styles={styles} isCollapsed={isCollapsed}>
           {icon && (
             <Icon
               icon={icon ? (iconArray ? icon[1] : icon) : null}
               prefix={icon && iconArray && icon[0]}
             />
           )}
-          {name}
+          {!isCollapsed && <span>{name}</span>}
         </StyledLink>
       </Link>
     )
   }
 
   return (
-    <StyledAside>
+    <StyledAside isCollapsed={isCollapsed}>
       <StyledBrand>
         <Image alt='Logo' src={brand} />
       </StyledBrand>
-
+      <div
+        style={{
```

Comment:
@esuarezz This is just nasty - we do not use Style in almost 99% of the components - why did you do here !?! This does annoy me to be fair as we have talked about this so many times. Give me a very good reason for this please.